File: travelAgency2/travelAgency2/Repository/LocationRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using travelAgency2.Repository.Database;
using travelAgency2.src.Domain;
using travelAgency2.src.Repository;
using travelAgency2.Utils;

namespace travelAgency2.Repository
{
    internal class LocationRepository : Repository<long, Location>
    {
        private Table<long, Location> table;

        public LocationRepository()
        {
            table = (LocationTable) Resources.getTableFactory().getTable(Constants.Db.Tables.LOCATION);
        }

        public void add(Location entity)
        {
            table.add(entity);
        }

        public void delete(long id)
        {
            throw new NotImplementedException();
        }

        public Location findByID(long id)
        {
            return table.findById(id);
        }

        public List<Location> getAll()
        {
            return table.getAll();
        }

        public void update(Location entity)
        {
            throw new NotImplementedException();
        }
    }
}

File: travelAgency/src/main/java/group/travelagency/repository/LocationRepository.java
package group.travelagency.repository;

import group.travelagency.domain.Location;
import group.travelagency.domain.dto.FlightDto;
import group.travelagency.repository.database.FlightTable;
import group.travelagency.repository.database.LocationTable;
import group.travelagency.repository.database.Table;
import group.travelagency.utils.Constants;
import group.travelagency.utils.Resources;

import java.sql.SQLException;
import java.util.List;
import java.util.Optional;

public class LocationRepository implements Repository<Long, Location> {
    private final Table<Long, Location> table;

    public LocationRepository() throws SQLException {
        table = (LocationTable) Resources.getTableFactory().getTable(Constants.Db.Tables.LOCATION);
    }


    @Override
    public void add(Location entity) {
        table.add(entity);
    }

    @Override
    public void update(Location entity, Location newEntity) {

    }

    @Override
    public Optional<Location> findByID(Long aLong) {
        return table.findById(aLong);
    }

    @Override
    public void delete(Long aLong) {
        //TODO
    }

    @Override
    public List<Location> getAll() {
        return table.getAll();
    }
}
